From 0b916306caf1a52b932d42c36ac5ea2d9213b0f5 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Tue, 18 Feb 2014 12:36:54 +0100
Subject: [PATCH 190/351] ASoC: arizona: Fix wrong number of items in enum
 ctls

arizona codec driver has a few places wrongly defining the number of
enum items.

Use SOC_ENUM_SINGLE_DECL() macro and they are automatically fixed.

[ Added in const before the SOC_ENUM_SINGLE_DECL definitions, this has
been added to the macro upstream but we need it manually added on this
kernel version - Charles ]

Change-Id: I06f799e62957eaf221252df08e4cade6bd55b9ad
Signed-off-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 sound/soc/codecs/arizona.c |   71 +++++++++++++++++++++++++-------------------
 1 file changed, 40 insertions(+), 31 deletions(-)

diff --git a/sound/soc/codecs/arizona.c b/sound/soc/codecs/arizona.c
index c2120c3..efe5bea 100644
--- a/sound/soc/codecs/arizona.c
+++ b/sound/soc/codecs/arizona.c
@@ -595,67 +595,76 @@ static const char *arizona_vol_ramp_text[] = {
 	"15ms/6dB", "30ms/6dB",
 };
 
-const struct soc_enum arizona_in_vd_ramp =
-	SOC_ENUM_SINGLE(ARIZONA_INPUT_VOLUME_RAMP,
-			ARIZONA_IN_VD_RAMP_SHIFT, 7, arizona_vol_ramp_text);
+const SOC_ENUM_SINGLE_DECL(arizona_in_vd_ramp,
+			   ARIZONA_INPUT_VOLUME_RAMP,
+			   ARIZONA_IN_VD_RAMP_SHIFT,
+			   arizona_vol_ramp_text);
 EXPORT_SYMBOL_GPL(arizona_in_vd_ramp);
 
-const struct soc_enum arizona_in_vi_ramp =
-	SOC_ENUM_SINGLE(ARIZONA_INPUT_VOLUME_RAMP,
-			ARIZONA_IN_VI_RAMP_SHIFT, 7, arizona_vol_ramp_text);
+const SOC_ENUM_SINGLE_DECL(arizona_in_vi_ramp,
+			   ARIZONA_INPUT_VOLUME_RAMP,
+			   ARIZONA_IN_VI_RAMP_SHIFT,
+			   arizona_vol_ramp_text);
 EXPORT_SYMBOL_GPL(arizona_in_vi_ramp);
 
-const struct soc_enum arizona_out_vd_ramp =
-	SOC_ENUM_SINGLE(ARIZONA_OUTPUT_VOLUME_RAMP,
-			ARIZONA_OUT_VD_RAMP_SHIFT, 7, arizona_vol_ramp_text);
+const SOC_ENUM_SINGLE_DECL(arizona_out_vd_ramp,
+			   ARIZONA_OUTPUT_VOLUME_RAMP,
+			   ARIZONA_OUT_VD_RAMP_SHIFT,
+			   arizona_vol_ramp_text);
 EXPORT_SYMBOL_GPL(arizona_out_vd_ramp);
 
-const struct soc_enum arizona_out_vi_ramp =
-	SOC_ENUM_SINGLE(ARIZONA_OUTPUT_VOLUME_RAMP,
-			ARIZONA_OUT_VI_RAMP_SHIFT, 7, arizona_vol_ramp_text);
+const SOC_ENUM_SINGLE_DECL(arizona_out_vi_ramp,
+			   ARIZONA_OUTPUT_VOLUME_RAMP,
+			   ARIZONA_OUT_VI_RAMP_SHIFT,
+			   arizona_vol_ramp_text);
 EXPORT_SYMBOL_GPL(arizona_out_vi_ramp);
 
 static const char *arizona_lhpf_mode_text[] = {
 	"Low-pass", "High-pass"
 };
 
-const struct soc_enum arizona_lhpf1_mode =
-	SOC_ENUM_SINGLE(ARIZONA_HPLPF1_1, ARIZONA_LHPF1_MODE_SHIFT, 2,
-			arizona_lhpf_mode_text);
+const SOC_ENUM_SINGLE_DECL(arizona_lhpf1_mode,
+			   ARIZONA_HPLPF1_1,
+			   ARIZONA_LHPF1_MODE_SHIFT,
+			   arizona_lhpf_mode_text);
 EXPORT_SYMBOL_GPL(arizona_lhpf1_mode);
 
-const struct soc_enum arizona_lhpf2_mode =
-	SOC_ENUM_SINGLE(ARIZONA_HPLPF2_1, ARIZONA_LHPF2_MODE_SHIFT, 2,
-			arizona_lhpf_mode_text);
+const SOC_ENUM_SINGLE_DECL(arizona_lhpf2_mode,
+			   ARIZONA_HPLPF2_1,
+			   ARIZONA_LHPF2_MODE_SHIFT,
+			   arizona_lhpf_mode_text);
 EXPORT_SYMBOL_GPL(arizona_lhpf2_mode);
 
-const struct soc_enum arizona_lhpf3_mode =
-	SOC_ENUM_SINGLE(ARIZONA_HPLPF3_1, ARIZONA_LHPF3_MODE_SHIFT, 2,
-			arizona_lhpf_mode_text);
+const SOC_ENUM_SINGLE_DECL(arizona_lhpf3_mode,
+			   ARIZONA_HPLPF3_1,
+			   ARIZONA_LHPF3_MODE_SHIFT,
+			   arizona_lhpf_mode_text);
 EXPORT_SYMBOL_GPL(arizona_lhpf3_mode);
 
-const struct soc_enum arizona_lhpf4_mode =
-	SOC_ENUM_SINGLE(ARIZONA_HPLPF4_1, ARIZONA_LHPF4_MODE_SHIFT, 2,
-			arizona_lhpf_mode_text);
+const SOC_ENUM_SINGLE_DECL(arizona_lhpf4_mode,
+			   ARIZONA_HPLPF4_1,
+			   ARIZONA_LHPF4_MODE_SHIFT,
+			   arizona_lhpf_mode_text);
 EXPORT_SYMBOL_GPL(arizona_lhpf4_mode);
 
 static const char *arizona_ng_hold_text[] = {
 	"30ms", "120ms", "250ms", "500ms",
 };
 
-const struct soc_enum arizona_ng_hold =
-	SOC_ENUM_SINGLE(ARIZONA_NOISE_GATE_CONTROL, ARIZONA_NGATE_HOLD_SHIFT,
-			4, arizona_ng_hold_text);
+const SOC_ENUM_SINGLE_DECL(arizona_ng_hold,
+			   ARIZONA_NOISE_GATE_CONTROL,
+			   ARIZONA_NGATE_HOLD_SHIFT,
+			   arizona_ng_hold_text);
 EXPORT_SYMBOL_GPL(arizona_ng_hold);
 
 static const char * const arizona_in_hpf_cut_text[] = {
 	"2.5Hz", "5Hz", "10Hz", "20Hz", "40Hz"
 };
 
-const struct soc_enum arizona_in_hpf_cut_enum =
-	SOC_ENUM_SINGLE(ARIZONA_HPF_CONTROL, ARIZONA_IN_HPF_CUT_SHIFT,
-			ARRAY_SIZE(arizona_in_hpf_cut_text),
-			arizona_in_hpf_cut_text);
+const SOC_ENUM_SINGLE_DECL(arizona_in_hpf_cut_enum,
+			   ARIZONA_HPF_CONTROL,
+			   ARIZONA_IN_HPF_CUT_SHIFT,
+			   arizona_in_hpf_cut_text);
 EXPORT_SYMBOL_GPL(arizona_in_hpf_cut_enum);
 
 static const char * const arizona_in_dmic_osr_text[] = {
-- 
1.7.9.5

